src/main/java/seedu/address/logic/parser/FindDeliveryCommandParser.java

```diff
@@ -0,0 +1,107 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATETIME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PRODUCT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_STATUS;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_SUPPLIER_INDEX;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import seedu.address.logic.commands.FindDeliveryCommand;
+import seedu.address.logic.commands.MarkSupplierCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.delivery.DateTime;
+import seedu.address.model.delivery.Delivery;
+import seedu.address.model.delivery.DeliveryDatePredicate;
+import seedu.address.model.delivery.DeliveryProductPredicate;
+import seedu.address.model.delivery.DeliveryStatusPredicate;
+import seedu.address.model.delivery.Status;
+import seedu.address.model.delivery.SupplierIndex;
+import seedu.address.model.product.Product;
+
+/**
+ * Parses input arguments and creates a new FindDeliveryCommand object.
+ */
+public class FindDeliveryCommandParser implements Parser<FindDeliveryCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindDeliveryCommand
+     * and returns a FindDeliveryCommand object for execution.
+     * @throws ParseException if the user input does not conform to the expected format.
+     */
+    public FindDeliveryCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_DATETIME, PREFIX_STATUS, PREFIX_PRODUCT, PREFIX_SUPPLIER_INDEX);
+
+        // Ensure that at least one filter is provided
+        if (!argMultimap.getValue(PREFIX_DATETIME).isPresent()
+                && !argMultimap.getValue(PREFIX_STATUS).isPresent()
+                && !argMultimap.getValue(PREFIX_PRODUCT).isPresent()
+                && !argMultimap.getValue(PREFIX_SUPPLIER_INDEX).isPresent()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindDeliveryCommand.MESSAGE_USAGE));
+        }
+
+        List<Predicate<Delivery>> predicates = new ArrayList<>();
+
+        // Parse the date if present
+        if (argMultimap.getValue(PREFIX_DATETIME).isPresent()) {
```

Comment:
There is a bug in the find delivery formula i believe because you have allowed multiple datetime/status/product prefixes. This leads to multiple values in the argMultiMap, however the get will only get the latest value given in the prefix. So the predicate only works on the last given prefix i think.

Reply:
I think there should be a check for duplicate prefixes and throw parseException if multiple same prefixes are found

Reply:
the check for duplicate prefixes has now been added

Reply:
Fixes #99